src/main/java/seedu/patientist/logic/parser/FindPatientCommandParser.java

```diff
@@ -0,0 +1,62 @@
+package seedu.patientist.logic.parser;
+
+import static seedu.patientist.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.patientist.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.patientist.logic.parser.CliSyntax.PREFIX_PID;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import seedu.patientist.logic.commands.FindPatientCommand;
+import seedu.patientist.logic.parser.exceptions.ParseException;
+import seedu.patientist.model.person.patient.PatientNameContainsKeywordsPredicate;
+import seedu.patientist.model.person.patient.PidContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all patients in patientist book whose name/pid contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindPatientCommandParser implements Parser<FindPatientCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindPatientCommand
+     * and returns a FindPatientCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindPatientCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PID);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PID) || !argMultimap.getPreamble().isEmpty()
+                || areBothPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PID)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+
+        Optional<String> name = argMultimap.getValue(PREFIX_NAME);
+        Optional<String> idNumber = argMultimap.getValue(PREFIX_PID);
+
+        if (idNumber.isEmpty()) {
+            String[] nameKeywords = name.get().trim().split("\\s+");
+            return new FindPatientCommand(new PatientNameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        } else {
+            String[] pidKeywords = idNumber.get().trim().split("\\s+");
+            return new FindPatientCommand(new PidContainsKeywordsPredicate(Arrays.asList(pidKeywords)));
+        }
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
```

Comment:
I think there's a slight bug in this header comment as the description seems to be for the method below.